File: tests/fixtures/java/basic/src/Service.java
package basic;

import java.util.List;
import java.util.ArrayList;

public interface Repository {
    List<String> findAll();
    String findById(String id);
}

class UserServiceImpl implements Repository {
    private List<String> users = new ArrayList<>();

    public UserServiceImpl() {
        users.add("Alice");
        users.add("Bob");
    }

    @Override
    public List<String> findAll() {
        return users;
    }

    @Override
    public String findById(String id) {
        return users.stream()
            .filter(u -> u.equals(id))
            .findFirst()
            .orElse(null);
    }

    public void addUser(String name) {
        users.add(name);
    }

    // Package-private method
    void internalProcess() {}
}

File: tests/fixtures/php/basic/service.php
<?php

namespace App;

interface Repository
{
    public function findAll(): array;
    public function findById(int $id): ?object;
}

class UserService implements Repository
{
    private array $users = [];

    public function __construct()
    {
        $this->users = ["Alice", "Bob"];
    }

    public function findAll(): array
    {
        return $this->users;
    }

    public function findById(int $id): ?object
    {
        return $this->users[$id] ?? null;
    }

    public function addUser(string $name): void
    {
        $this->users[] = $name;
    }

    // Protected method
    protected function validate(string $name): bool
    {
        return !empty($name);
    }

    // Private method
    private function internalProcess(): void {}
}
